openedx/features/discounts/docs/decisions/0001-determining-discount-location.rst

```diff
@@ -37,14 +58,21 @@ Decisions
    b. Outside of the LMS, we need a way to get this information in a validated fashion. We have added a rest 
       endpoint for that communication. Because we want to be able to use the information both on the front-end
       and back-end without risk of the user creating their own discounts, we are sending it as a jwt, which is
-      signed. See C for more information on how specifically it will be used on the ecommerce front-end.
+      signed. See C for more information on how specifically it will be used on the Ecommerce front-end.
 
-   c. We do not want to make a synchronous call from ecommerce to the LMS every time the basket page is loaded.
-      However, we are happy to make an ajax call from the front-end.
+   c. We do not want to make a synchronous call from the Ecommerce backend to the LMS every time the basket 
+      page is loaded. See this decision record link for more information: 
+      https://github.com/edx/Ecommerce/blob/master/docs/decisions/0002-no-synchronous-calls.rst  However, 
+      we are happy to make an ajax call from the front-end.
       The basket page needs this information for two reasons: 1) So that the correct price is shown to the user,
       and the user is informed that they are recieving a discount. 2) So that the correct price is charged to 
-      the user during payment.  We will hand both by making the ajax call, updating the ui on the page, and also
-      updating the data that is sent to the server during the payment step. Because we are using a JWT, we can
-      verify that the discount is correct. Right now, another team is working on adding a MFE to ecommerce. 
-      Rather than have to implement the front-end code in ecommerce to do this twice, we will write the code
-      to make the ajax call, and if there is a discount, reload the page passing the JWT as a parameter.
+      the user during payment.  We will handle both by making the ajax call, updating the ui on the page, and
```

Comment:
is the assumption that the async call will be on the front end or the backend?
I understand that the decision record linked above assumes we can just do ajax calls and update the necessary portion of the page.  
However, currently applying coupons requires a full page reload, not sure what would be the case with discounts.  
We definitely wouldn't want a front-end async call towards the end of the page load to trigger a full page reload - this would effectively be a synchronous call that also doubles the load time of the page

Reply:
We are okay with it as an interim step, before the MFE is built.

Reply:
how would the MFE solve this issue if coupons and offers are applied on the backend?

Reply:
The application of the discount happens during the purchase. The MFE will allow us to easily display the discount information prior to purchase, making it unnecessary to re-load the page with that information.

Reply:
i know that coupons get applied at a basket level, i think the same is true of enterprise offers?
do you mean that we're proposing a change to the behavior for this new type of discount where they will only be applied at the order level?  
that seems like a big change, which maybe deserves its own small section in this document  
not sure how this would affect our reporting, oscar extensions, basket code logic, etc...

Reply:
Matthew and I talked offline -- the TL;DR of that convo is that the MFE will still communicate the updated basket information to the backend while loading that page.